src/main/java/com/amihaiemil/docker/RtContainer.java

```diff
@@ -38,7 +47,32 @@
 
     @Override
     public JsonObject inspect() throws IOException {
-        return null;
+        final HttpGet inspect = new HttpGet(this.baseUri.toString() + "/json");
+        final HttpResponse response = this.client.execute(inspect);
+        final JsonObject info;
+        if(response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+            info = Json
+                .createReader(response.getEntity().getContent()).readObject();
+        } else {
+            info = null;
```

Comment:
@amihaiemil why are we returning `null`? Why not throw an error? Or why not have the return type be `Optional<JsonObject>`?

I really think we should throw an error. At the very least, not return `null` in any case.